### PR TITLE
fix flushBackoffQCompleted: remove defer

### PR DESCRIPTION
**Pull request by duc-cnzj**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
resource leak.

#### Which issue(s) this PR fixes:

fix resource leak in loop.

https://github.com/kubernetes/kubernetes/blob/3f6793de4862647fc221f4acf9d61adace73a592/pkg/scheduler/internal/queue/scheduling_queue.go#L410-L433

remove defer

https://github.com/kubernetes/kubernetes/blob/3f6793de4862647fc221f4acf9d61adace73a592/pkg/scheduler/internal/queue/scheduling_queue.go#L431

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/internal/queue/scheduling_queue.go [hunk 1]
```diff
@@ -428,7 +428,7 @@ func (p *PriorityQueue) flushBackoffQCompleted() {
 		}
 		p.activeQ.Add(rawPodInfo)
 		metrics.SchedulerQueueIncomingPods.WithLabelValues("active", BackoffComplete).Inc()
-		defer p.cond.Broadcast()
+		p.cond.Broadcast()
```
Comment (ahg-g) [inline pkg/scheduler/internal/queue/scheduling_queue.go:431]: I would change the returns above to breaks, add a flag to indicate whether or not to broadcast and move the broadcast call outside the loop.

Comment (DuC-cnZj) [reply pkg/scheduler/internal/queue/scheduling_queue.go:436]: @ahg-g  Thanks for the tip, updated

**Review by binacs — APPROVED** [review]
/ok-to-test

**Pull request merged**